### PR TITLE
Implement support for custom Api Versions in informers (#639)

### PR DESCRIPTION
**Pull request by cambierr**
This adds the support of custom api versions to the informer library as discussed in #639

### CODE REVIEW

**Comment by graemerocher** [comment]
is it possible to add a test?

**Comment by cambierr** [comment]
done !

File: kubernetes-client/src/main/java/io/micronaut/kubernetes/client/ModelMapper.java [hunk 1]
```diff
@@ -38,12 +41,17 @@
...
+    private static final Logger LOG = LoggerFactory.getLogger(ModelMapper.class);
+
     // Model's api-group prefix to kubernetes api-group
     private final Map<String, String> preBuiltApiGroups = new HashMap<>();
 
     // Model's api-version midfix to kubernetes api-version
     private final List<String> preBuiltApiVersions = new ArrayList<>();
 
+    // This allows parsing custom (not included in kubernetes core) api versions
+    private final Pattern customVersionParser = Pattern.compile("(V[a-z1-9]+)[A-Z]+[a-zA-Z0-9]+");
```
Comment (n0tl3ss) [inline kubernetes-client/src/main/java/io/micronaut/kubernetes/client/ModelMapper.java]: Did you get this regex from somewhere or have you created it? 

I saw that you said this are valid:
- V1MySettings
- V1alpha1Operation

what about: 
- Vtest1Operation

Comment (cambierr) [reply kubernetes-client/src/main/java/io/micronaut/kubernetes/client/ModelMapper.java]: I actually created it myself based on all the CRDs I know from my previous experiences but looking slightly more for references, I just found this one: https://github.com/kubernetes/apimachinery/blob/master/pkg/util/version/version.go#L38

Do you think we should use it ?

Comment (cambierr) [reply kubernetes-client/src/main/java/io/micronaut/kubernetes/client/ModelMapper.java]: Any update @n0tl3ss ?

Comment (n0tl3ss) [reply kubernetes-client/src/main/java/io/micronaut/kubernetes/client/ModelMapper.java]: Hi @cambierr can you update the PR with the regex that you have mentioned.

Comment (cambierr) [reply kubernetes-client/src/main/java/io/micronaut/kubernetes/client/ModelMapper.java]: sure, done!

**Review by n0tl3ss — APPROVED** [review]

**Comment by graemerocher** [comment]
Thanks for the contribution!

**Pull request merged**